### PR TITLE
[WIP]support size for mark-tick

### PR DESCRIPTION
**Pull request by YuhanLu**
i did the similar thing as mark bar, but still not sure if this what we expected about size of mark-tick? #932

### CODE REVIEW

File: test/compile/mark-tick.test.ts [hunk 1]
```diff
@@ -68,4 +68,22 @@ describe('Mark: Tick', function() {
       assert.deepEqual(props.yc, {scale: Y, field: 'Cylinders'});
     });
   });
+
+  describe('with size.value specified with orient vertical', function() {
```
Comment (kanitw) [inline test/compile/mark-tick.test.ts]: You should test when you map a field to `size` rather than value. 

Also, add another test for horizontal case.

**Comment by kanitw** [comment]
The logic in mark-tick.ts looks correct.  Please finish the test and ping us ;)

**Comment by YuhanLu** [comment]
Unit tests added @kanitw

File: test/compile/mark-tick.test.ts [hunk 2]
```diff
@@ -68,4 +68,40 @@ describe('Mark: Tick', function() {
...
+      'encoding':
+        {
+          'x': {'field': 'Horsepower', 'type': 'quantitative'},
+          'y': {'field': 'Cylinders','type': 'ordinal'},
+          'size': {'field': 'Acceleration'}
+        },
+      'data': {'url': 'data/cars.json'},
+    });
+    const props = tick.properties(model);
+    it('width should change with size', function() {
```
Comment (kanitw) [inline test/compile/mark-tick.test.ts]: width should be mapped to size.

File: test/compile/mark-tick.test.ts [hunk 3]
```diff
@@ -68,4 +68,40 @@ describe('Mark: Tick', function() {
...
```
Comment (kanitw) [inline test/compile/mark-tick.test.ts]: height should be mapped to size.

File: test/compile/mark-tick.test.ts [hunk 4]
```diff
@@ -68,4 +68,40 @@ describe('Mark: Tick', function() {
...
+  });
+
+  describe('with size.value specified with orient vertical', function() {
+    const model = parseModel({
+      'mark': 'tick',
+      'config': {'mark': {'orient': 'horizontal'}},
+      'encoding':
+        {
+          'x': {'field': 'Horsepower', 'type': 'quantitative'},
+          'y': {'field': 'Cylinders','type': 'ordinal'},
```
Comment (kanitw) [inline test/compile/mark-tick.test.ts]: try to add space after comma for readability.

File: test/compile/mark-tick.test.ts [hunk 5]
```diff
@@ -68,4 +68,40 @@ describe('Mark: Tick', function() {
...
+        },
+      'data': {'url': 'data/cars.json'},
+    });
+    const props = tick.properties(model);
+    it('width should change with size', function() {
+      assert.deepEqual(props.width, {'field': 'Acceleration', 'scale': SIZE});
+    });
+  });
+
+  describe('with size.value specified with orient vertical', function() {
```
Comment (kanitw) [inline test/compile/mark-tick.test.ts]: "with size.value" => "with size's field"

File: test/compile/mark-tick.test.ts (hunk 1 above)
Comment (kanitw) [inline test/compile/mark-tick.test.ts]: "with size.value" => "with size's field"

**Comment by YuhanLu** [comment]
fixed those unit tests @kanitw

File: test/compile/mark-tick.test.ts [hunk 6]
```diff
@@ -68,4 +68,40 @@ describe('Mark: Tick', function() {
...
+      'encoding':
+        {
+          'x': {'field': 'Horsepower', 'type': 'quantitative'},
+          'y': {'field': 'Cylinders', 'type': 'ordinal'},
+          'size': {'field': 'Acceleration'}
+        },
+      'data': {'url': 'data/cars.json'},
+    });
+    const props = tick.properties(model);
+    it('width should change with size field', function() {
```
Comment (kanitw) [inline test/compile/mark-tick.test.ts]: height :)

**Comment by YuhanLu** [comment]
fixed again @kanitw

**Comment by kanitw** [comment]
:tada:

**Pull request merged**